streaming/src/test/scala/org/apache/spark/streaming/CheckpointSuite.scala

```diff
@@ -644,8 +644,10 @@ class CheckpointSuite extends TestSuiteBase with DStreamCheckpointTester
         val mappedStream = fileStream.map(s => {
           val i = s.toInt
           if (i == 3) {
-            while (CheckpointSuite.batchThreeShouldBlockIndefinitely) {
-              Thread.sleep(Long.MaxValue)
+            if (CheckpointSuite.batchThreeShouldBlockIndefinitely) {
+              // It's not a good idea to let the thread run forever
+              // as resource won't be correctly released
+              Thread.sleep(6000)
```

Comment:
@srowen It's a bit hacky here, but I can't find a good way. The problem here is that, if the Task here waits indefinitely, and the `reader` in the `NewHadoopRDD` will never get a chance to close. And the file handle held by it cannot be deleted by windows.